spicy/toolchain/src/compiler/codegen/parsers/literals.cc

```diff
@@ -163,11 +162,72 @@ struct Visitor : public hilti::visitor::PreOrder<Expression, Visitor> {
                 return dst;
             }
 
-            case LiteralMode::Try:
-                // RegExp are special-cased by the parser generator and handled
-                // directly there, so that we should get here in try mode.
-                hilti::logger().internalError("unexpected literal mode Try when parsing regexp literals");
-            default: hilti::util::cannot_be_reached();
+            case LiteralMode::Search: // Handled in `parseLiteral`.
+            case LiteralMode::Try: {
```

Comment:
The code in this case block is almost identical the `Default` case: let's factor that out into a function or something, it's quite substantial code duplication.